pilot/pkg/features/pilot.go

```diff
@@ -704,6 +704,9 @@ var (
 
 	StackdriverAuditLog = env.Register("STACKDRIVER_AUDIT_LOG", false, ""+
 		"If enabled, StackDriver audit logging will be enabled.").Get()
+
+	EnableGatewayMtlsDownstreamMetadataExchnage = env.RegisterBoolVar("ENABLE_GATEWAY_MTLS_DOWNSTREAM_METADATA_EXCHANGE", false,
```

Comment:
probably we do not need this flag

Reply:
I am personally in favour of enabling this by default without any feature flag. I added the feature flag considering the issues highlighted here: https://github.com/istio/istio/pull/48141/#issuecomment-1839883860.

Let me know if removing the feature flag and enabling it by default should be safe.